src/features/Chat/components/ActiveWindow/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,121 @@
+import { useTranslation } from 'react-i18next';
+
+import {
+  Buddy,
+  ChatFolder,
+  useUpdateStatusMutation,
+} from '@/features/Chat/chatPageApi';
+import { useAppSelector } from '@/store';
+import { selectUserId } from '@/features/Authentication/userSlice';
+
+import styled from 'styled-components';
+import { IconButton, TextButton } from '@/components/Buttons';
+import IconWarning from '@/static/icons/warning.svg';
+import { palette } from '@/components/variables';
+import Text from '@/components/Text';
+
+export type DialogVariant = 'archive' | 'block' | 'restore';
+
+type Props = {
+  variant: DialogVariant;
+  chat: Buddy;
+  close: () => void;
+};
+
+const ConfirmationDialog = ({ variant, chat, close }: Props) => {
+  const { t } = useTranslation('chat');
+
+  const [updateChatStatus] = useUpdateStatusMutation();
+  const userId = useAppSelector(selectUserId);
+
+  const variants: Record<
+    DialogVariant,
+    { borderColor: string; targetFolder: ChatFolder }
+  > = {
+    archive: { borderColor: palette.orange, targetFolder: 'archived' },
+    block: { borderColor: palette.redSalmon, targetFolder: 'banned' },
+    restore: { borderColor: palette.blue2, targetFolder: 'ok' },
+  };
+
+  const confirm = () => {
+    close();
+    if (!userId) return;
+    const status: ChatFolder = variants[variant].targetFolder;
```

Comment:
Is it necessary to assert the type?

Reply:
Just a clarification, maybe not necessary. 🤔